src/libcryptobox/cryptobox.c

```diff
@@ -697,7 +712,8 @@ bool rspamd_cryptobox_verify(const unsigned char *sig,
 		g_assert(EVP_DigestSignInit(sha_ctx, &pctx, EVP_sha512(),
 									ENGINE_F_ENGINE_NEW, pkey) == 1);
 
-		g_assert(EVP_DigestSignFinal(sha_ctx, sig, &siglen) == 1);
+		unsigned char non_const_sig = *sig;
+		g_assert(EVP_DigestSignFinal(sha_ctx, &non_const_sig, &siglen) == 1);
```

Comment:
On the contrary, it is **possible** to convert `const char *` to `unsigned char *`, and with your change the whole signatures checking will be broken, as apparently signature is not a 1 character string (it also will cause OOB reading which is a potential vulnerability).